sw/device/silicon_creator/lib/drivers/uart.c

```diff
@@ -97,14 +97,21 @@ int uart_getchar(uint32_t timeout_ms) {
   return n ? (int)ch : -1;
 }
 
-size_t uart_write(const uint8_t *data, size_t len) {
-  size_t total = len;
+void uart_write(const void *data, size_t len) {
+  const uint8_t *d = (const uint8_t *)data;
```

Comment:
we should add some of the countermeasures we discussed offline to avoid enabling an attacker to dump the flash by manipulating the stack via faults. 

`data` should only point to ro in ROM or ROM_EXT, and `len` should be limited to the limit of ro. 

We can add a `public_strings` region if you want to make this more portable.

Reply:
Rather than putting section limits on `uart_write`, I went the route of encoding the string as little-endian bytes in a uint64 so the whole chunk could be passed by value rather than by reference.

This means editing the banner message is somewhat tedious, but we all can find an ASCII table when we need one.

Reply:
ack. That makes sense. I have been using https://github.com/cfrantz/asciitable :)